src/soc/ibm/power9/mvpd.c

```diff
@@ -62,9 +62,9 @@ struct pt_record {
 
 /* Reads from a single EEPROM chip, which is deduced from offset. Returns number
  * of bytes read. */
-static int read_eeprom_chip(uint32_t offset, void *data, uint16_t len)
+static int read_eeprom_chip(uint8_t cpu, uint32_t offset, void *data, uint16_t len)
 {
-	const unsigned int bus = 1;
+	const unsigned int bus = 1 + 3 * cpu; // three I2C buses per CPU
```

Comment:
It should be `4`, I noticed it when adding SPD on second CPU. It works with `3` too because it's not the zeroes engine and all checks evaluate the same. Will update.